src/resmom/linux/alps.c

```diff
@@ -5110,213 +5892,424 @@ alps_cancel_reservation(job *pjob)
 
 /**
  * @brief
- *	 Issue an ENGINE query and determine which version of BASIL
- * 	we should use.
+ *	 Issue a request to switch an existing reservation "OUT" (suspend it)
+ * or "IN" (resume it).
  *
- * @return Void
+ * Called by mother superior when a job needs to be suspended or resumed.
  *
+ * @retval 0 success
+ * @retval 1 transient error (retry)
+ * @retval -1 fatal error
  */
-static void
-alps_engine_query(void)
+int
+alps_suspend_resume_reservation(job *pjob, basil_switch_action_t switchval)
```

Comment:
This function seems outside the scope of using the new BASIL 1.3/1.4 interface to fill in information for resources PBS already has.  I don't think this code should be added.